File: LeetCodeChallenges/Arrays 101/MergeSortedArray.java
public class MergeSortedArray {
    public static void main(String[] args) {
        int[] num1 = { 1, 2, 3, 0, 0, 0 };
        int m = 3;

        int[] num2 = { 2, 5, 6 };
        int n = 3;

        merge(num1, m, num2, n);
    }

    public static void sort(int[] arr) {
        int n = arr.length;
        int temp = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 1; j < (n - i); j++) {
                if (arr[j - 1] > arr[j]) {
                    // swap elements
                    temp = arr[j - 1];
                    arr[j - 1] = arr[j];
                    arr[j] = temp;
                }

            }
        }

    }

    public static void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] tempArr = new int[m + n];
        int index = 0;
        int counter = 0;
        boolean isOn = true;
        while (isOn) {
            if (nums1[counter] != 0) {
                tempArr[index] = nums1[counter];
                index++;
            }
            if (nums2[counter] != 0) {
                tempArr[index] = nums2[counter];
                index++;
            }
            counter++;
            if (index == tempArr.length) {
                isOn = false;
            }
        }
        sort(tempArr);
    }
}

File: Arrays101/Inserting Item Into Array/mergesortedarray.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        if n == 0:
            return nums1
        else:
            counter = 0
            for x in range(m + n):
            
                if counter < n:
                    nums1[m + x] = nums2[counter]
                    counter = counter +1
            nums1.sort()